service/generation.js

```diff
@@ -0,0 +1,119 @@
+const fs = require("fs");
+const { spawn } = require("child_process");
+
+const logger = require("tslog");
+const appLogger = new logger.Logger({
+    minLevel: 1,
+});
+
+function generate(data, requestId, contractTemplate, progressConnection) {
+    if (data.mediaContractualObjects) {
+        return generateContract(data, requestId, contractTemplate, progressConnection);
+    } else if (data.turtle?.value) {
+        return generateFromTurtle(data.turtle.value, requestId, contractTemplate, progressConnection);
+    } else {
+        throw new Error("Turtle nor MCO found on the data payload");
+    }
+}
+
+// generates and creates and contract from turtle specification
+function generateFromTurtle(turtleSpec, requestId, contractTemplate, progressConnection) {
+    appLogger.debug("Generation from Turtle file");
+    appLogger.debug(`Contract template to use: '${contractTemplate}'`);
+    appLogger.debug(`Request Id: '${requestId}'`);
+
+    const baseFile = `${requestId}-${Date.now()}`;
+
+    // First of all we need to take the Turtle and transform to MCO
+    const path = `.contracts/turtle/${baseFile}.ttl`;
+    fs.writeFileSync(path, turtleSpec);
+
+    const outputDir = `service/.contracts/mco`;
+    const mcoParserDir = `../MPEG-21-Part-23-MCO-Parser`;
+
+    const mcoFileName = `${baseFile}-mco.json`;
+
+    const generationCommand = `npx ts-node ${mcoParserDir}/test/index.js -c "service/${path}" > ${outputDir}/${mcoFileName}`;
+
+    appLogger.debug("From Turtle generation cmd", generationCommand);
+
+    const ps = spawn("sh", ["-c", generationCommand], { cwd: ".." });
+
+    ps.stdout.on("data", (data) => {
+        console.log(`stdout: ${data}`);
+        if (progressConnection) {
+            progressConnection.send(`${data}`);
+        }
+    });
+
+    ps.stderr.on("data", (data) => {
+        console.error(`stderr: ${data}`);
+        if (progressConnection) {
+            progressConnection.send("<error>");
+            progressConnection.send(`${data}`);
+            progressConnection.send("</error>");
+        }
+    });
+
+    ps.on("close", (code) => {
+        appLogger.debug(`MCO generation process exited with code ${code}`);
+        if (code !== 0) {
+            appLogger.error("There was an error generating the MCO objects");
+        } else {
+            appLogger.debug("Now generating the contract and writing it to the Ledger");
+            generateFromMcoFile(mcoFileName, requestId, contractTemplate, progressConnection);
+        }
+    });
+}
```

Comment:
Why the use of the command from console? I think it is easier to maintain using the package itself:

```javascript
const { getContractFromMCO } = require('mco-parser');

const ttl = fs.readFileSync(contractPath, 'utf-8');
const res = await getContractFromMCO(ttl);
```